docs/distribution/launcher.md

```diff
@@ -0,0 +1,188 @@
+The command-line allows to edit project configuration, for example: change the
+author's name or Enso version.
+
+## Enso and Graal Version Management
+The launcher automatically manages required Enso versions. When running inside a
+project, if the version specified in project configuration is not installed, it
+is installed automatically.
+
+If an Enso version is no longer needed, it can be removed with the `uninstall`
+command.
+
+Moreover, GraalVM distributions tied to the installed Enso versions are managed
+automatically by the launcher. When a new Enso version is installed, it also
+ensures that the correct GraalVM version is installed and that it is used for
+launching this version of Enso. When a managed GraalVM distribution is no longer
+used by any installed version of Enso, it is automatically removed.
+
+### GraalVM Override
+While the launcher manages its own installation of GraalVM to ensure that the
+right JVM version is used to launch each version of Enso, the user can override
+this mechanism to use the installed system JVM instead. This is an advanced
+feature and should rarely be used.
+
+The launcher will check the system JVM and refuse to launch Enso if it is not a
+GraalVM distribution. It will also print a warning if the major version is
+different then required by that particular Enso version.
+
+### Downloading Enso Releases
+The releases are discovered and downloaded using the
+[GitHub API](https://docs.github.com/en/rest/reference/repos#releases). As
+described in the [Release Policy](./release-policy.md#github-releases), each
+release contains a Manifest file that is downloaded first. It specifies if this
+Enso version can be used with the current launcher or an upgrade is needed, as
+described in
+[Minimal Required Launcher Version](#minimal-required-launcher-version). If the
+version is correct, the binary file containing the Enso components distribution
+is downloaded. The Manifest also specifies which GraalVM version should be used
+with this version of Enso. If that version of GraalVM is not present on the
+system it is also installed.
+
+Releases [marked as broken](./release-policy.md#marking-a-release-as-broken) are
+ignored by the launcher unless it is specified by an exact version match. In
+that case it is downloaded, but a warning is printed.
+
+### Downloading GraalVM Releases
+GraalVM is downloaded from its
+[GitHub releases page](https://github.com/graalvm/graalvm-ce-builds/releases)
+using GitHub API, similarly as Enso releases.
+
+> The actionables for this section are:
+> 
+> - Should the launcher also allow to manage versions of Graal EE in the future?
+
+## Running Enso Components
+The primary purpose of the launcher is running various Enso components, namely
+the REPL, running a project, Enso scripts or the language server.
+
+The launcher automatically infers which Enso version to used, based on the
+project configuration. If used outside an Enso project, the `default` version
```

Comment:
Outside a project or running a script.

Reply:
I assume that when inside a project, even if a script is run, the project version should be used (a bit how `jenv` selects the Java version). At least that is my intuition of what a user might expect, but I do understand that different users may have different expectations.

Do you think it's enough to document this behaviour in the README / `--help` ?

Reply:
If my working directory is inside a project and I run a script at a relative path outside the project I want that to use my global configured version. If I run a script where the script file is located inside the current project then yes it should use the project version.

Reply:
I updated the description to say that for scripts, the script path is used to infer the version (ie. if the script is located inside a project, version from this project's configuration is used): https://github.com/enso-org/enso/pull/963/files#diff-21ea226f235b522c88fc903583e8ec10R122-R132